manifests/build-support/mkDerivationBasic.nix

```diff
@@ -33,5 +33,5 @@ stdenv.mkDerivation {
     ln -s $BIN $out/bin/${name}
   '';
 
-  meta = meta lib;
+  meta = meta lib // {mainProgram = name;};
```

Comment:
Is this record merge necessary if `meta` is already setting `mainProgram = pname;`? This looks like a redundant merge to me.

Reply:
The meta function didn't set `mainProgram`. Hence, there was this merge. I feel it's better to set `mainProgram` where `mkDerivationBasic.nix` function is called (09bdc089de03a81825f1a6dae8ec30fafa351aeb) because then,`mainProgram` can be changed independently of `name`.